tests/unit/test_task_proxy.py

```diff
@@ -101,3 +103,27 @@ def test_status_match(status_str: Optional[str], expected: bool):
     mock_itask = Mock(state=Mock(status='waiting'))
 
     assert TaskProxy.status_match(mock_itask, status_str) is expected
+
+
+@pytest.mark.parametrize('itask_flow_nums, flow_nums, expected', [
+    param({1, 2}, {2}, {2}, id="subset"),
+    param({2}, {1, 2}, {2}, id="superset"),
+    param({1, 2}, {3, 4}, set(), id="disjoint"),
+    param({1, 2}, set(), {1, 2}, id="all-matches-num"),
+    param(set(), {1, 2}, set(), id="num-doesnt-match-none"),
+    param(set(), set(), set(), id="all-doesnt-match-none"),
+])
+def test_match_flows(
+    itask_flow_nums: FlowNums, flow_nums: FlowNums, expected: FlowNums
+):
+    mock_itask = Mock(flow_nums=itask_flow_nums)
+    assert TaskProxy.match_flows(mock_itask, flow_nums) == expected
+
+
+def test_match_flows_copy():
+    """Test that this method does not return the same reference as
+    itask.flow_nums."""
```

Comment:
Worth stating why it's particularly important that it doesn't return the same ref?

Reply:
Added to https://github.com/MetRonnie/cylc-flow/pull/34